include/aie/Dialect/AIE/IR/AIEOps.td

```diff
@@ -1610,7 +1610,10 @@ def AIE_ObjectFifoCreateOp: AIE_Op<"objectfifo", [HasParent<"DeviceOp">, Symbol]
         TypeAttrOf<AIE_ObjectFifoType>:$elemType,
         BDDimLayoutArrayAttr:$dimensionsToStream,
         BDDimLayoutArrayArrayAttr:$dimensionsFromStreamPerConsumer,
-        DefaultValuedAttr<BoolAttr, "false">:$via_DMA
+        DefaultValuedAttr<BoolAttr, "false">:$via_DMA,
+        // via_shared_mem==0 means use producer tile's memory module
```

Comment:
As I understand this, the goal is to control the buffer placement.  However a buffer could be shared but not in the producer or consumer (i.e. in a column, a producer in row 2 communicates to a consumer in row 4 through a buffer in row 3.)  So is this really sufficient?  At some level, even in a non-shared memory situation we have the same question, as to where buffers are allocated.  Perhaps we should really be looking into how to solve some of these buffer placement questions more holistically?

Reply:
I fully agree with you Steve. There are a lot of scenarios in which we could use the shared memory module of another tile rather than the one in the producer / consumer. At the moment the lowering will always allocate objects in the producer's shared memory module and doesn't do any analysis to figure out whether that is the best choice. I definitely want to move towards a more general solution in the near future.

The addition in this PR is a momentary solution to at least be able to choose between two available shared memory modules in a column, as we've encountered that situation is Gagan's designs and this addition would enable us to progress faster. However, we don't need to merge this PR into main now. I can keep working on it to cover more cases.